src/borg/archive.py

```diff
@@ -81,15 +86,36 @@ def __add__(self, other):
         stats.osize_parts = self.osize_parts + other.osize_parts
         stats.usize_parts = self.usize_parts + other.usize_parts
         stats.nfiles_parts = self.nfiles_parts + other.nfiles_parts
+        stats.chunking_time = self.chunking_time + other.chunking_time
+        stats.hashing_time = self.hashing_time + other.hashing_time
+        for key in other.files_stats.keys():
```

Comment:
cosmetic: the code as you wrote it works, but even more elegant is to use the default behaviour:
```
for key in some_dict:
```
That automatically iterates over the keys in some_dict by default.